modules/core/src/smithy4s/http/internals/package.scala

```diff
@@ -63,12 +63,17 @@ package object internals {
   }
 
   private def fromToString(str: String): Option[PathSegment] = {
-    if (str.isEmpty()) None
-    else if (str.startsWith("{") && str.endsWith("+}"))
-      Some(PathSegment.greedy(str.substring(1, str.length() - 2)))
-    else if (str.startsWith("{") && str.endsWith("}"))
-      Some(PathSegment.label(str.substring(1, str.length() - 1)))
-    else Some(PathSegment.static(str))
+    Option(str).map { str =>
```

Comment:
I think you got rid of a `.isEmpty()` here

you may want to reintroduce it as a `.filter(_.nonEmpty)` on the `Option`

Reply:
ah yes , was thinking of Null , will update